tests/libnet/cloudinit/configdrive/network_data.go

```diff
@@ -0,0 +1,148 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2023 Red Hat, Inc.
+ *
+ */
+
+package configdrive
+
+import "encoding/json"
+
+// LinkType represents interface type
+type LinkType string
+
+const Phy LinkType = "phy"
+
+// Link represents L2 interface settings
+type Link struct {
+	Id                 string   `json:"id"`
+	Type               LinkType `json:"type"`
+	EthernetMacAddress string   `json:"ethernet_mac_address"`
+	MTU                int      `json:"mtu,omitempty"`
+}
+
+// Route represents L3 IPv4 / IPv6 routing configuration item
+type Route struct {
+	Network string `json:"network"`
+	Netmask string `json:"netmask"`
+	Gateway string `json:"gateway"`
+}
+
+type NetworkType string
+
+const (
+	IPv4     NetworkType = "ipv4"
+	IPv4DHCP NetworkType = "ipv4_dhcp"
+	IPv6     NetworkType = "ipv6"
+	IPv6DHCP NetworkType = "ipv6_dhcp"
+)
+
+// Network represents L3 network
+type Network struct {
+	Id        string      `json:"id"`
+	Type      NetworkType `json:"type"`
+	Link      string      `json:"link"`
+	NetworkId string      `json:"network_id"`
+	IPAddress string      `json:"ip_address,omitempty"`
+	Netmask   string      `json:"netmask,omitempty"`
+	Routes    []Route     `json:"routes,omitempty"`
+}
+
+type NetworkOption func(network *Network)
+
+func NewNetwork(id string, Type NetworkType, link string, options ...NetworkOption) Network {
+	network := Network{
+		Id:   id,
+		Type: Type,
+		Link: link,
+	}
+
+	for _, option := range options {
+		option(&network)
+	}
+
+	return network
+}
+
+func WithIPAddress(IPAddress string) NetworkOption {
+	return func(network *Network) {
+		network.IPAddress = IPAddress
+	}
+}
+
+func WithNetmask(netmask string) NetworkOption {
+	return func(network *Network) {
+		network.Netmask = netmask
+	}
+}
+
+func WithRoute(route Route) NetworkOption {
```

Comment:
Couldn't find any usage of it in the PR, I would remove it.

Reply:
Removed.